test_conformance/math_brute_force/main.c

```diff
@@ -1349,6 +1383,44 @@ test_status InitCL( cl_device_id device )
 #endif
                 }
 #endif /* __APPLE__ */
+
+                gFloatToHalfRoundingMode = kRoundToNearestEven;
+                if (strstr(ext, "cl_khr_fp16"))
```

Comment:
Suggest you use `is_extension_available` and rework the existing code. This would reduce indentation in this code quite a bit and could be done as a pre-requisite patch.

Reply:
In my opinion the rework should be done in the whole test suit (not only this part for half). I suggest to move this work to separate issue. Please define new github issue if you think this is important.